### PR TITLE
Add support to custom sorting in storage index search

### PR DESCRIPTION
**Pull request by hexun80149128**
https://forum.heroiclabs.com/t/does-the-query-in-storageindexlist-support-sorting-and-how/4948
 currently storage index listing does not support sorting the results.so add it。thanks

### CODE REVIEW

**Comment by ftkg** [comment]
We probably also want to index the fields with the `Sortable()` property before doing requests with custom sorts, as specified in the [docs](https://blugelabs.com/bluge/sorting/).

**Comment by hexun80149128** [comment]
RegisterStorageIndex add sortFields param.it must be in fields,and should be string or num. @ftkg

File: server/storage_index.go [hunk 1]
```diff
@@ -483,6 +485,28 @@ func (si *LocalStorageIndex) mapIndexStorageFields(userID, collection, key, vers
 	rv.AddField(bluge.NewNumericField("read", float64(read)).StoreValue())
 	rv.AddField(bluge.NewNumericField("write", float64(write)).StoreValue())
 
+	if len(sortFilters) > 0 {
```
Comment (ftkg) [inline server/storage_index.go:488]: This is duplicating the indexing of values that happens in `BlugeWalkDocument`. You could possibly set the `Sortable` property directly there, but keep in mind that other features also depend on this function.

Comment (hexun80149128) [reply server/storage_index.go]: after BlugeWalkDocument set the Sortable property by sortFilters,and the sortFilters must start with "value.". I didn't modify BlugeWalkDocument as other features also depend on this function.

Comment (hexun80149128) [reply server/storage_index.go]: @ftkg

Comment (ftkg) [reply server/storage_index.go]: Thanks @hexun80149128 we'll evaluate how best to arrange this part of the code before merging it.

Comment (ftkg) [reply server/storage_index.go:488]: @hexun80149128 can you please sign up the CLA so we can accept this PR later?

**Comment by ftkg** [comment]
`nakama-common` PR: https://github.com/heroiclabs/nakama-common/pull/142

File: server/storage_index.go (hunk 1 above)
Comment (hexun80149128) [reply server/storage_index.go]: @ftkg done.

**Comment by hexun80149128** [comment]
@ftkg can merge to master next version？

**Review by sesposito — APPROVED** [review]

**Comment by sesposito** [comment]
@ftkg please add a changelog entry as well.

**Pull request merged**